projects/ng-time-chart/src/lib/layout/layout-selector.service.ts

```diff
@@ -0,0 +1,41 @@
+import {Injectable} from '@angular/core';
+import {StackedLayoutService} from './stacked-layout.service';
+import {TiledLayoutService} from './tiled-layout.service';
+import {Item} from '../item';
+
+export enum LayoutStrategy {
+  /**
+   * Stack items one on top of the other
+   */
+  Stacked,
+  /**
+   * Try to tile items if their start and end dates do not overlap
+   */
+  Tiled
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class LayoutSelectorService {
```

Comment:
This service and the switch case would not be necessary, if the `[layoutStrategy]` would accept a `LayoutService`. The using component could then simply :tumbler_glass: let inject the desired strategy and pass it using to the time chart component. As a side effect, mocking for `NgTimeChartComponent` tests would be easier.